Client/src/Components/Graph/Graph.tsx

```diff
@@ -31,19 +35,74 @@ export default function Graph(props: IProps) {
   const outerHeight = props && props.outerHeight || 500
   const width = outerWidth - margin.left - margin.right
   const height = outerHeight - margin.top - margin.bottom
+  const extremeBoundaries = props.extremeBoundaries
 
   const [isXLog, setXToggle] = useState(false)
   const [isYLog, setYToggle] = useState(false)
+  const [showSettings, setSettingsToggle] = useState(false)
+  const [xUpperBound, setXUpperBound] = useState(extremeBoundaries.maxX)
+  const [xLowerBound, setXLowerBound] = useState(extremeBoundaries.minX)
+  const [yUpperBound, setYUpperBound] = useState(extremeBoundaries.maxY)
+  const [yLowerBound, setYLowerBound] = useState(extremeBoundaries.minY)
+  // This loops through the selected arrays to find the min/max for both x and y.
 
   const handleXScaleClick = () => {
+    if (xLowerBound <= 0 && !isXLog) {
+      setXLowerBound(1)
+    }
     setXToggle(!isXLog)
   }
 
   const handleYScaleClick = () => {
+    if (yLowerBound <= 0 && !isYLog) {
+      setYLowerBound(1)
+    }
     setYToggle(!isYLog)
   }
 
-  let active = [null, null, null, null]
+  const handleSettingsClick = () => {
+    setSettingsToggle(!showSettings)
+  }
+
+  const handleXUpperBoundChange = (event) => {
+    const { value } = event.target
+    if ((value < extremeBoundaries.maxX)) {
+      setXUpperBound(Math.ceil(extremeBoundaries.maxX))
+    }
+    else {
+      setXUpperBound(value)
+    }
+  }
+  const handleXLowerBoundChange = (event) => {
+    const { value } = event.target
+    if ((value <= 0 && isXLog) || (value > extremeBoundaries.minX)) {
+      setXLowerBound(Math.floor(extremeBoundaries.minX))
+    }
+    else {
+      setXLowerBound(value)
+    }
+  }
+
+  const handleYUpperBoundChange = (event) => {
```

Comment:
you could actually have this all done as a formik form and then you wouldnt need to write your own handlers

Reply:
I'll try to modify it to use Formik instead.

Reply:
I decided not to use Formik due to it working oddly when trying to override the FastField with an onchange prop.